Fix composite compliance tensor for inplace views

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #79767
* __->__ #79902

Fixes: https://github.com/pytorch/pytorch/issues/79748

We shouldn't be running behavior keys again on the wrapper here, because we'd be running them twice. It also breaks the composite compliance metadata check for forward AD when both the primal and tangent are CCT.

soulitzer committed:
Fix composite compliance tensor for inplace views

[ghstack-poisoned]

soulitzer committed:
Update on "Fix composite compliance tensor for inplace views"


Fixes: https://github.com/pytorch/pytorch/issues/79748

We shouldn't be running behavior keys again on the wrapper here, because we'd be running them twice. It also breaks the composite compliance metadata check for forward AD when both the primal and tangent are CCT.

[ghstack-poisoned]

zou3519 reviewed (approved):
thanks! test failures seem unrelated?

soulitzer committed:
Update on "Fix composite compliance tensor for inplace views"


Fixes: https://github.com/pytorch/pytorch/issues/79748

We shouldn't be running behavior keys again on the wrapper here, because we'd be running them twice. It also breaks the composite compliance metadata check for forward AD when both the primal and tangent are CCT.

[ghstack-poisoned]

soulitzer commented:
Oh test failures were real, a composite compliance test that was xfailed is now passing (it was doing in-place views)
